app/services/mastodon_tasks_service.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+class MastodonTasksService
+  def self.daily
+    log_start_task('feeds::clear')
+    Feeds.clear
+    log_end_task('feeds::clear')
+
+    log_start_task('media::clear')
+    Media.clear
+    log_end_task('media::clear')
+
+    log_start_task('users::clear')
+    Users.clear
+    log_end_task('users::clear')
+
+    log_start_task('push::refresh')
+    Push.refresh
+    log_end_task('push::refresh')
+  end
+
+  class Media
+    class << self
+      def clear
+        MediaAttachment.where(status_id: nil).where('created_at < ?', 1.day.ago).find_each(&:destroy)
+      end
+
+      def remove_silenced
+        MediaAttachment.where(account: Account.silenced).find_each(&:destroy)
+      end
+
+      def remove_remote
+        MediaAttachment.where.not(remote_url: '').where('created_at < ?', 1.week.ago).find_each do |media|
```

Comment:
If you do move these to class methods, there are probably some existing scopes you can re-use to make this easier to read.